polaris/ocean/rpe.py

```diff
@@ -0,0 +1,109 @@
+import csv
+
+import numpy as np
+import xarray as xr
+from mpas_tools.cime.constants import constants
+
+
+def compute_rpe(initial_state_file_name, output_files):
+    """
+    Computes the reference (resting) potential energy for the whole domain
+
+    Parameters
+    ----------
+    initial_state_file_name : str
+        Name of the netCDF file containing the initial state
+
+    output_files : list
+        List of netCDF files containing output of forward steps
+
+    Returns
+    -------
+    rpe : numpy.ndarray
+        the reference potential energy of size ``Time`` x ``len(output_files)``
+    """
+
+    gravity = constants['SHR_CONST_G']
+
+    dsInit = xr.open_dataset(initial_state_file_name)
+    nVertLevels = dsInit.sizes['nVertLevels']
+
+    xEdge = dsInit.xEdge
+    yEdge = dsInit.yEdge
+    areaCell = dsInit.areaCell
+    minLevelCell = dsInit.minLevelCell - 1
+    maxLevelCell = dsInit.maxLevelCell - 1
+    bottomDepth = dsInit.bottomDepth
+
+    areaCellMatrix = np.tile(areaCell, (nVertLevels, 1)).transpose()
+    bottomMax = np.max(bottomDepth.values)
+    yMin = np.min(yEdge.values)
+    yMax = np.max(yEdge.values)
+    xMin = np.min(xEdge.values)
+    xMax = np.max(xEdge.values)
+    areaDomain = (yMax - yMin) * (xMax - xMin)
+
+    vert_index = \
+        xr.DataArray.from_dict({'dims': ('nVertLevels',),
+                                'data': np.arange(nVertLevels)})
+
+    cell_mask = np.logical_and(vert_index >= minLevelCell,
+                               vert_index <= maxLevelCell)
+    cell_mask = np.swapaxes(cell_mask, 0, 1)
+
+    num_files = len(output_files)
+
+    if num_files == 0:
+        raise ValueError('Must provide at least one output filename')
```

Comment:
Let's move this check to the top of the routine.

Reply:
Yes, I'll move it to the top.